tests/Engagement_unit_test.py

```diff
@@ -0,0 +1,141 @@
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement updated successfully.', EngagementTXT))
+
+    def test_engagement_import_scan_result(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_class_name("pull-left").click()
+        driver.find_element_by_link_text('View Engagements').click()
+        driver.find_element_by_link_text("edited test engagement").click()
+        driver.find_elements_by_id("dropdownMenu1")[1].click()
+        driver.find_element_by_link_text('Import Scan Results').click()
+        driver.find_element_by_id('id_active').get_attribute('checked')
+        driver.find_element_by_id('id_verified').get_attribute('checked')
+        Select(driver.find_element_by_id("id_scan_type")).select_by_visible_text("Nmap Scan")
+        driver.find_element_by_id('id_file').send_keys(os.getcwd().replace('/tests', '') + "/dojo/unittests/scans/nmap_sample/nmap_multiple_port.xml")
+        driver.find_element_by_css_selector("input.btn.btn-primary").click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Nmap Scan processed, a total of 13 findings were processed', EngagementTXT))
+
+    def test_close_new_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_class_name("pull-left").click()
+        driver.find_element_by_link_text("View Engagements").click()
+        driver.find_element_by_link_text("edited test engagement").click()
+        driver.find_element_by_id("dropdownMenu1").click()
+        driver.find_element_by_link_text("Close Engagement").click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement closed successfully.', EngagementTXT))
+
+    def test_delete_new_closed_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_class_name("pull-left").click()
+        driver.find_element_by_link_text('View Engagements').click()
+        driver.find_element_by_link_text("edited test engagement").click()
+        driver.find_element_by_id("dropdownMenu1").click()
+        driver.find_element_by_link_text('Delete Engagement').click()
+        driver.find_element_by_name('delete_name').click()
+        EngagementTXT = driver.find_element_by_tag_name("BODY").text
+        self.assertTrue(re.search(r'Engagement and relationships removed.', EngagementTXT))
+
+    def test_new_ci_cd_engagement(self):
+        driver = self.login_page()
+        driver.get(self.base_url + "product")
+        driver.find_element_by_link_text('Engagement Test').click()
+        driver.find_element_by_xpath("//a[@class='dropdown-toggle active']//span[@class='hidden-xs']").click()
```

Comment:
This line gave me issues. Commenting it allowed the test to run to success. Not sure it's necessary.

Reply:
yes but that click on import result but I want to click on Done button to separate add engagement test.

Reply:
This is due to this issue https://github.com/DefectDojo/django-DefectDojo/issues/1160. The version without the pretty UI doesn't have a drop down. Everything is plainly listed. Hold tight for now.